apps/dashboard/app/controllers/files_controller.rb

```diff
@@ -114,7 +114,11 @@ def normalized_path
 
   def show_file
     if params[:download]
-      send_file @path
+      if Files.downloadable?(@path)
+        send_file @path
+      else
+        render json: { error_message: "#{@path} exceeds the maximum configured download size #{ ::Configuration.file_download_max } bytes" }, status: :forbidden
```

Comment:
Shouldn't this be status 413 payload too large?

Reply:
@ericfranz Which do you prefer? HTTP 403 and HTTP 413 can both work here

Reply:
403 - The server understood the request, but will not fulfill it.
413 - The request entity is larger than limits defined by server.